tools/latest-kernel-full-config.sh

```diff
@@ -121,20 +121,95 @@ inner_full_config() {
         export KCONFIG_CONFIG=bottlerocket_${arch}_defconfig
 
         br_cfg="${kernel_path}/config-bottlerocket"
+        br_cfg_arch="${kernel_path}/config-bottlerocket-${arch}"
         microcode_cfg="/microcode/${microcode_file}"
 
         pushd "linux-${version}" || bail "Could not move into linux-${version}"
 
         if [ "${arch}" = "aarch64" ]; then
             karch="arm64"
-            script_args=("../config-${arch}" "${br_cfg}")
+            script_args=("../config-${arch}" "${br_cfg}" "${br_cfg_arch}")
         elif [ "${arch}" = "x86_64" ]; then
             karch="x86"
-            script_args=("../config-${arch}" "${microcode_cfg}" "${br_cfg}")
+            script_args=("../config-${arch}" "${microcode_cfg}" "${br_cfg}" "${br_cfg_arch}")
         fi
 
         ARCH=${karch} ./scripts/kconfig/merge_config.sh "${script_args[@]}"
         mv -f "bottlerocket_${arch}_defconfig" "${kernel_path}/config-full-bottlerocket-${arch}" || bail "Failed to create config-full-bottlerocket-${arch}"
         popd || bail "Could not move around - 'popd' failed in merge_config loop. Lets stop before we break anything further."
     done
+
+    # Validate the generated configurations
+    echo "Validating generated kernel configurations..."
+    validate_kernel_configs
+}
+
+# Function to validate kernel configurations (similar to validate_config.sh)
+validate_kernel_configs() {
+    local success=true
```

Comment:
It's kind of cute, but I'd avoid doing this in bash as it may not be obvious that the value assigned to this variable is ultimately executed as a command.

I tend to initialize to zero and increment (via `(( ++errors ))`), then check for an error count greater than zero at the end.